context/app/static/js/components/cells/CellTypeDistributionChart/CellTypeDistributionChart.tsx

```diff
@@ -0,0 +1,261 @@
+      },
+      {
+        counts: [] as number[],
+        otherLabels: [] as string[],
+        targetLabels: [] as string[],
+        total: 0,
+        countsMap: {} as Record<string, number>,
+      },
+    );
+    return {
+      ...results,
+      sortedData: sorted,
+    };
+  }, [data, targetCellTypes]);
+
+  const scale = useLinearScale(cellCounts, {
+    range: [0, parentWidth!], // range to normalize the values to percentages
+    domain: [0, totalCellCount],
+  });
+
+  const xOffsets = useMemo(() => {
+    const results = cellCounts.reduce(
+      ({ currentOffset, offsets }, currentValue) => {
+        offsets.push(currentOffset);
+        const newOffset: number = currentOffset + (scale(currentValue) as number);
+        return { currentOffset: newOffset, offsets };
+      },
+      {
+        currentOffset: 0,
+        offsets: [] as number[],
+      },
+    );
+    return results.offsets;
+  }, [cellCounts, scale]);
+
+  const targetCellColors = useChartPalette();
+  const targetColorScale = useOrdinalScale(presentTargetCellTypes, {
+    domain: presentTargetCellTypes,
+    range: targetCellColors,
+  });
+  const otherColorScale = useOrdinalScale(otherCellLabels, {
+    domain: otherCellLabels,
+    range: unselectedCellColors,
+  });
+
+  const {
+    tooltipData,
+    tooltipLeft,
+    tooltipTop,
+    tooltipOpen,
+    containerRef,
+    TooltipInPortal,
+    handleMouseEnter,
+    handleMouseLeave,
+  } = useChartTooltip<TooltipData<CellTypeCountForTissue>>();
+
+  return (
+    <>
+      <Typography variant="subtitle2" component="label" color="primary" mb={1}>
+        Cell Type Distribution Across {capitalizeString(tissue)} Datasets
+      </Typography>
+      <svg direction="row" width={parentWidth} height="50" ref={containerRef}>
+        {sortedData.map(({ cell_count, index: label }, index) => {
+          const isTargetedCellType = targetCellTypes.includes(label);
+          return (
+            <rect
+              key={label}
+              width={scale(cell_count) as number}
+              x={xOffsets[index]}
+              height={isTargetedCellType ? 40 : 30}
+              y={isTargetedCellType ? 5 : 10}
+              fill={isTargetedCellType ? targetColorScale(label) : otherColorScale(label)}
+              // Highlight the bar if it is currently being hovered for the tooltip
+              filter={
+                label === tooltipData?.key ? 'brightness(120%) drop-shadow(0px 0px 5px rgba(0, 0, 0, 0.5))' : undefined
+              }
+              style={{ transition: 'filter 0.2s ease-in-out' }}
+              onMouseEnter={handleMouseEnter({
+                key: label,
+                bar: { data: { cell_count, index: label } },
+              })}
+              rx={isTargetedCellType ? 5 : 0}
+              ry={isTargetedCellType ? 5 : 0}
+              onMouseLeave={handleMouseLeave}
+              aria-label={`${label}: ${cell_count} (${((cell_count / totalCellCount) * 100).toFixed(2)}%)`}
```

Comment:
Given the number of ternaries, it could be a bit cleaner to add a util which returns the props.